libimage/disk_usage.go

```diff
@@ -31,12 +33,19 @@ type ImageDiskUsage struct {
 // storage.  Note that a single image may yield multiple usage reports, one for
 // each repository tag.
 func (r *Runtime) DiskUsage(ctx context.Context) ([]ImageDiskUsage, int64, error) {
-	images, err := r.ListImages(ctx, nil, nil)
+	storageListResult, err := r.store.MultiList(
+		storage.MultiListOptions{Images: true, Layers: true},
+	)
 	if err != nil {
 		return nil, -1, err
 	}
 
-	layerTree, err := r.layerTree(ctx, images)
+	images := make([]*Image, 0, len(storageListResult.Images))
+	for i := range storageListResult.Images {
+		images = append(images, r.storageToImage(&storageListResult.Images[i], nil))
```

Comment:
(Very vaguely, intuitively it seems to me that a loop turning `storage.Image` into `libimage.Image` should not need to be open-coded. But that’s an imprecise hunch and we can revisit after correctness and code structure settles.)